### DISCUSSION TITLE
v2.2.4

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#3096] [Releases]
## Installation
Visit [https://getutm.app/install/][1] for the most up to date installation instructions.

## New Features (Summary)
* (macOS 11.3+ Only) **Bridged networking and shared networking support**. You can configure it in the Network settings for your VM.
* (macOS 11+ and iOS 13+) **GPU acceleration for OpenGL on Linux**. Use `virtio-ramfb-gl` or `virtio-vga-gl` display device and compatible Linux drivers (most modern Linux distros will have it already installed). Windows is not supported because there is currently no virtio-gpu driver for Windows that supports 3D acceleration. Note that newly created VMs will default to a "GPU Supported" display device on supported architectures but existing VMs must manually change the display device in Display settings. GPU acceleration is still an experimental feature, so it may not work in some situations (including many 3D use cases).
* **EFI Boot**. By default new VMs created for pc, q35, and virt* machines will have EFI enabled. In previous versions, EFI is only enabled for virt* machines. Due to compatibility with boot, existing pc and q35 VMs will NOT have EFI enabled and must be manually turned on in Settings -> System -> Advanced Configuration -> UEFI Boot. This may also require you to re-install the bootloader on your VM. As part of this change, EFI variables will also be properly handled (on both ARM and x86 VMs). Note that if you've configured a custom pflash device for your VM, the new UEFI Boot option will take no effect (same as before).
* **QEMU v6.1.0** is now used for the backend.

## Changes (v2.2.1)
* Fix crash when setting custom VM icon (#2387) (thanks @ktprograms)
* Disable Port Forwarding feature for bridged networking (thanks @conath)
* Fixed memory leak (#2720)

## Changes (v2.2.2)
* (iOS) Enable OpenGL acceleration for iOS 11-12
* (iOS 15+) Increased memory limit for supported devices (M1 iPads currently)
* (macOS) Fix layout issues in settings on macOS Monterey (#2644)
* (macOS) Fix capturing of hotkeys (such as Cmd+Tab) (#2677)
* Disable port forwarding options for shared & bridged networking
* Fixed crash when PPC screamer is selected on iOS (currently this is still not supported)
* Refactored non-OpenGL rendering code, fixing some random crashes seen in TestFlight reports.

## Changes (v2.2.3)
* (iOS) When a mouse/trackpad is already connected when UTM starts up, it will now be properly detected (#3013)
* (iOS) Handle touch and Apple Pencil input while the mouse is captured (#2669)
* Windows BSOD on boot/setup due to `PAGE_FAULT_IN_NONPAGED_AREA` (#2721)

## Changes (v2.2.4)
* (macOS) Settings no longer crash when an error occurs while saving a new VM
* (iOS) GL rendering now works on EFI startup (black screen during boot) (#3076)

## Known Issues
* v2.2.4 is built with Xcode 12.5.1 instead of Xcode 13.0 beta due to Apple's strict requirements in submitting apps built on beta SDKs. As a result, the layout issues in settings on macOS Monterey are back in v2.2.4. This will be resolved when Xcode 13 is released.
* Networking can be broken in some instances, we are still investigating the cause. (#3051, #3094)

Please check the full list on [Github][2] and help report any bug you find that is not listed.

[1]: https://getutm.app/install/
[2]: https://github.com/utmapp/UTM/issues

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v2.2.4'>v2.2.4</a>.</em>

**Comment by conath** [comment]
I have tested the following guests; host mentioned and system architecture is in braces:

## M1 MacBook Pro on macOS 11.5.2

**Existing VMs:**  
- Mac OS 9 (PowerPC) OK
- DOS (i368 / x86) OK
- Windows 11 for ARM (aarch64 with Hypervisor; "Shared Network" mode; `virtio-gpu-pci` with DOD drivers) OK
 => #2628 persists, as expected, just like #2729
- Ubuntu 20.04 LTS ARM OK: Desktop works, 3D crashes as expected #3053
- ReactOS (x86_64) OK
- Windows 10 21H1 (x86_64) OK

**Installers:**  
- Ubuntu ARM (aarch64 hvf) OK in console and graphics modes; #3094 as expected for Ubuntu guest
- Windows 10 (x86_64) OK
- Windows 11 (aarch64) OK

## MacBook Air (Intel) on macOS 11.5.2

Tested HVF by installing a new Linux Mint VM with default x86_64 settings (EFI, shared network); works fine. 

## iPad Pro A12X on iOS 14.7.1

**Existing VMs:**  
- Windows 10 (aarch64) FAIL (stuck at EFI booting screen; no Windows spinner)
- Windows 10 (x86_64) OK
- Windows 8.1 32-bit (x86_64) OK

**Installers:**
- Ubuntu 20.04 Server ARM (aarch64) OK

**Comment by Daitien97** [comment]
> ## Installation
> 
> Visit [https://getutm.app/install/][1] for the most up to date installation instructions.
> 
> 
> 
> ## New Features (Summary)
> 
> * (macOS 11.3+ Only) **Bridged networking and shared networking support**. You can configure it in the Network settings for your VM.
> 
> * (macOS 11+ and iOS 13+) **GPU acceleration for OpenGL on Linux**. Use `virtio-ramfb-gl` or `virtio-vga-gl` display device and compatible Linux drivers (most modern Linux distros will have it already installed). Windows is not supported because there is currently no virtio-gpu driver for Windows that supports 3D acceleration. Note that newly created VMs will default to a "GPU Supported" display device on supported architectures but existing VMs must manually change the display device in Display settings. GPU acceleration is still an experimental feature, so it may not work in some situations (including many 3D use cases).
> 
> * **EFI Boot**. By default new VMs created for pc, q35, and virt* machines will have EFI enabled. In previous versions, EFI is only enabled for virt* machines. Due to compatibility with boot, existing pc and q35 VMs will NOT have EFI enabled and must be manually turned on in Settings -> System -> Advanced Configuration -> UEFI Boot. This may also require you to re-install the bootloader on your VM. As part of this change, EFI variables will also be properly handled (on both ARM and x86 VMs). Note that if you've configured a custom pflash device for your VM, the new UEFI Boot option will take no effect (same as before).
> 
> * **QEMU v6.1.0** is now used for the backend.
> 
> 
> 
> ## Changes (v2.2.1)
> 
> * Fix crash when setting custom VM icon (#2387) (thanks @ktprograms)
> 
> * Disable Port Forwarding feature for bridged networking (thanks @conath)
> 
> * Fixed memory leak (#2720)
> 
> 
> 
> ## Changes (v2.2.2)
> 
> * (iOS) Enable OpenGL acceleration for iOS 11-12
> 
> * (iOS 15+) Increased memory limit for supported devices (M1 iPads currently)
> 
> * (macOS) Fix layout issues in settings on macOS Monterey (#2644)
> 
> * (macOS) Fix capturing of hotkeys (such as Cmd+Tab) (#2677)
> 
> * Disable port forwarding options for shared & bridged networking
> 
> * Fixed crash when PPC screamer is selected on iOS (currently this is still not supported)
> 
> * Refactored non-OpenGL rendering code, fixing some random crashes seen in TestFlight reports.
> 
> 
> 
> ## Changes (v2.2.3)
> 
> * (iOS) When a mouse/trackpad is already connected when UTM starts up, it will now be properly detected (#3013)
> 
> * (iOS) Handle touch and Apple Pencil input while the mouse is captured (#2669)
> 
> * Windows BSOD on boot/setup due to `PAGE_FAULT_IN_NONPAGED_AREA` (#2721)
> 
> 
> 
> ## Changes (v2.2.4)
> 
> * (macOS) Settings no longer crash when an error occurs while saving a new VM
> 
> * (iOS) GL rendering now works on EFI startup (black screen during boot) (#3076)
> 
> 
> 
> ## Known Issues
> 
> * v2.2.4 is built with Xcode 12.5.1 instead of Xcode 13.0 beta due to Apple's strict requirements in submitting apps built on beta SDKs. As a result, the layout issues in settings on macOS Monterey are back in v2.2.4. This will be resolved when Xcode 13 is released.
> 
> * Networking can be broken in some instances, we are still investigating the cause. (#3051, #3094)
> 
> 
> 
> Please check the full list on [Github][2] and help report any bug you find that is not listed.
> 
> 
> 
> [1]: https://getutm.app/install/
> 
> [2]: https://github.com/utmapp/UTM/issues
> 
> <hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v2.2.4'>v2.2.4</a>.</em>
